### DISCUSSION TITLE
Install with proxy that requires each site be approved

### DISCUSSION THREAD

**Discussion by steelerguy** [Security-Onion-Solutions/securityonion#3553] [Unsupported Versions] [answered]
I recently installed 2.3.30 on a machine that has internet access via a proxy, but needs to have each site whitelisted in the proxy. I had this working with 16.04. I could do soup, get Ubuntu packages and docker images. I used the ISO to install 2.3.30 and there were no proxy configuration options in the installer. I went to a shell and configured the proxy, but got errors trying to pull packages for Cent OS (expected, I don't know wheat sites I need to access for) and github (not expected, I have at least some access). The install looked like it was working, but would freeze at 80% trying to start The Hive. I could not get past this point.

I then did an install and chose airgapped and this worked. Once complete, I configured the proxy and tried to pull new ET Pro rules, it looked locally for them which makes sense for an airgap install. I found the setting and set airgap to false and did a salt highstate. This really seemed to break things, still could not pull the rules (I could use wget to download them), and seemed to have incorrect URLs.

So this leads me to a couple questions. Is there a way to install through a proxy that needs to have each site specifically white listed? A list of sites in the documentation would be nice rather than having to look through failure logs or asking the maintainer of the proxy to look through logs to see what lookups failed.  Next question, is there a way to "convert" an airgap install to a non-airgap setup after installation?

I really wanted to get a working setup, so I am remaining in airgap state and not trying to configure internet because I am afraid of blowing something up and having to reinstall which takes a good bit of time (at least having to use virtual media as I am not onsite). I'll attempt to maintain it like this, but would like to just be able to run soup and having rules update automatically.

**Comment by TOoSmOotH** [comment, marked as answer]
2.3.40 will have better proxy support and should allow you to run an install and update rules using a proxy server. As far as what it reaches out to in a non-airgap is as follows:

CentOS/Ubuntu Repos
Docker Repo
Saltstack Repo
Wazuh Repo
Epel Repo (For CentOS only)
Github for soup and Strelka signatures
ghcr.io for Docker images (Github Images Repo)
AWS S3 for Docker Image Signatures
Emerging Threats for Suricata rules
